src/rubrix/server/elasticseach/client_wrapper.py

```diff
@@ -102,7 +102,8 @@ def list_documents(
         A sequence of documents resulting from applying the query on the index
 
         """
-        return es_scan(self.__client__, query=query or {}, index=index)
+        # Preserve order is used to ensure that the samples are sorted by identifier
+        return es_scan(self.__client__, query=query or {}, index=index, preserve_order=True)
```

Comment:
I've been doing some tests and I think this should generally work. We avoid using scroll and use the `search_after` mechanism to iterate the whole dataset:

```suggestion
        size = size or 1000
        query = query.copy() or {}
        query["sort"] = [{"_id": {"order": "asc"}}] # Force sorting by id
        response = self.__client__.search(index=index, body=query, size=size)
        while response["hits"]["hits"]:
            for hit in response["hits"]["hits"]:
                yield hit

            last_id = hit["_id"]
            query["search_after"] = [last_id]
            response = self.__client__.search(index=index, body=query, size=size)
```